lib/markdown.js

```diff
@@ -1,35 +1,45 @@
 /* jshint node:true */
 "use strict";
 
+var rHeader = /^\s*(\#+|={3,}|-{3,})/;
+var rBadge = /^\[!\[[^\]]+\]\([^)]+\)\]\([^)]+\)$/;
```

Comment:
This currently doesn't match an image with no alt text `[![](image-src)](some-link)` but that could be fixed up with `*` instead of `+` -> `^\[!\[[^\]]*\]\([^)]*\)\]\([^)]*\)$`. Let's add a test for this 

URLs can contain paranthesis, `https://en.wikipedia.org/wiki/Saw_(disambiguation)` which is another pitfall of this regex. Perhaps we need to use something like https://www.npmjs.com/package/balanced-match

Reply:
Nice catch! I will replace `+` with `*` and add a test. For URLs that contain parentheses, how about simply removing the last `$`? It is not the faultless solution, but must work in most circumstances.

I just want to keep this package as zero-dependency. If don't mind, then I will try `balanced-match`. Or copy the main logic into this package.

Reply:
Adding a dependency for proper parsing is fine with me